# HG changeset patch
# User Rik <rik@octave.org>
# Date 1514323923 28800
#      Tue Dec 26 13:32:03 2017 -0800
# Node ID a94a1d742938b31745478b7618a0749892c71362
# Parent  536926c2975c16f0d490a5e8cf166120b1a653e2
getappdata.m: Use try/catch for 40% performance improvement.

* getappdata.m: Replace complicated if/else testing with
try/catch block.

diff --git a/scripts/miscellaneous/getappdata.m b/scripts/miscellaneous/getappdata.m
--- a/scripts/miscellaneous/getappdata.m
+++ b/scripts/miscellaneous/getappdata.m
@@ -43,23 +43,20 @@ function value = getappdata (h, name)
   endif
 
   if (nargin == 2)
     if (! ischar (name))
       error ("getappdata: NAME must be a string");
     endif
 
     value = cell (numel (h), 1);
-    appdata = struct ();
     for i = 1:numel (h)
-      value{i} = [];
-      pval = get (h(i));
-      if (isfield (pval, "__appdata__") && isfield (pval.__appdata__, name))
-        value{i} = pval.__appdata__.(name);
-      endif
+      try
+        value{i} = (get (h(i), "__appdata__")).(name);
+      end_try_catch
     endfor
 
     if (i == 1)
       value = value{1};
     endif
 
   else  # nargin == 1
     if (numel (h) != 1)
